### ISSUE TITLE
Incrementation doesn't work for <table>

### ISSUE THREAD

**Issue by razbounak** [ninsuo/symfony-collection#74]
Hi, 

When I add a <tr> element in my custom form theme, the incrementation of my collection doesn't works !

```
{% block PriceType_row %}
    {{ form_widget(form) }}
{% endblock %}

{% block PriceType_widget %}
    <tr>
        <td>
            {{ form_widget(form.quantityFrom, {'attr': {'class': 'form-control'}}) }}
        </td>
        <td>
            {{ form_widget(form.quantityTo, {'attr': {'class': 'form-control'}}) }}
        </td>
        <td>
            {{ form_widget(form.bestUnitPrice, {'attr': {'class': 'form-control'}}) }}
        </td>
        <td>
            {{ form_widget(form.getPaid, {'attr': {'class': 'form-control'}}) }}
        </td>
        <td>
            {{ form_widget(form.vatRate, {'attr': {'class': 'form-control'}}) }}
        </td>
    </tr>
{% endblock %}

{% block ProductType_widget %}
    {{ form_widget(form.prices) }}l
{% endblock %}
```

Thanks for your help ! :)

**Comment by ninsuo** [comment]
Indeed.
I will check that tomorrow night.
Thanks!

**Comment by ninsuo** [comment]
Just for my information, did you try to change `elements_selector` setting to `> tr`?

**Comment by razbounak** [comment]
Where ? 

$('.collection-prices').collection({
    /* Here ? */
});

**Comment by ninsuo** [comment]
exactly.

```js
$('.collection-prices').collection({
   elements_selector: '> tr',
});
```

**Comment by razbounak** [comment]
OMG This seems to work !!!
Where is elements_selector in the doc ?

**Comment by ninsuo** [comment]
I was looking for it too, it seems that I forgot it.
I leave this issue open to add the doc and add a demo with a table.

**Comment by razbounak** [comment]
Thank you very mush @ninsuo for all your work !

**Comment by razbounak** [comment]
Can I use a bettor selector ?
My collection is just between <div class="table-responsive"> and <table>


```
                                    <div class="table-responsive">
                                        <table class="table table-stripped table-bordered">
                                            <thead>
                                                <tr>
                                                    <th>
                                                        Quantity from
                                                    </th>
                                                    <th>
                                                        Quantity to
                                                    </th>
                                                    <th>
                                                        Best unit price
                                                    </th>
                                                    <th>
                                                        You get paid
                                                    </th>
                                                    <th>
                                                        VAT rate
                                                    </th>
                                                    <th>
                                                        Actions
                                                    </th>
                                                </tr>
                                            </thead>
                                            <tbody id="prices_table">
                                                <div id="collect-prices">
                                                    {{ form_row(form.prices) }}
                                                </div>
                                            </tbody>
                                        </table>

                                        <a href="" data-collection="collection-prices" class="collection-action collection-add btn btn-success">Add</a>

                                        <div class="text-right">
                                            <button type="submit" class="btn btn-primary">
                                                <i class="fa fa-check"></i> SAVE
                                            </button>
                                        </div>

                                    </div>
```

**Comment by ninsuo** [comment]
Sorry but I don't understand your question.

You say `My collection is just between <div class="table-responsive"> and <table>` but in your example I see your form below `<div id="collect-prices">`

**Comment by razbounak** [comment]
Yes, It's true
I don't understand why the collection is here and not in my <div id="collect-prices">

**Comment by ninsuo** [comment]
Ah ok, so you mean that your collection is visually located above the `<table>` instead of being below `<div id="collect-prices">` ?

**Comment by razbounak** [comment]
Yes

**Comment by ninsuo** [comment]
Ok, will look forward tonight.

**Comment by ninsuo** [comment]
It's a bit late so it will wait tomorrow! But I don't forget it 👍

**Comment by razbounak** [comment]
Thank you :)

**Comment by ninsuo** [comment]
Yep, I can reproduce your issue.

This is because Symfony generates some markup (like divs etc) around the collection itself, so your browser displays it above the table.

So your form theme should not only contain a theme for each collection item, it should also overwrite your collection type itself. 

It makes sense, your table is only used by the collection itself, so it should be the collection's form theme.

I'll push my demo within a few minutes (making it more fancy), it'll make things clearer 👍

**Comment by razbounak** [comment]
Thanks a lot Ninsuo, I'll check your demo

**Comment by ninsuo** [comment]
Here you go: 
https://symfony-collection.fuz.org/symfony3/basic/inATable

I noticed a bug anyway with the "add" button located at the bottom (add_at_the_end set to true or all items removed from the collection): the add button at the bottom will add elements below the collection container (the <table> in our case), so at the same deepness as `<tbody>` instead of below. 

I will not be able to programmatically fix it because collections may be created with 0 elements, so I can't guess the right place using one existing item. But I'll add an option, creating the issue now and closing this one.

Enjoy!

**Comment by razbounak** [comment]
I have a problem with your new demo : 
jquery.collection.js: given collection selector does not exist.

```
$('.table collection-prices').collection({
                elements_selector: 'tr.price',
                add_at_the_end: true,
                custom_add_location: true,
            });
```

```
{% block collection_widget %}
    {% spaceless %}

    {#
     # This is almost a copy/paste of jquery.collection.html.twig, we can't use it as it also
     # use form_widget. Note that we also use form_widget(prototype) instead of form_row(prototype)
     # to avoid generating extra markup.
     #}
    {% if prototype is defined %}
        {% set attr = attr|merge({'data-prototype': form_widget(prototype)}) %}
        {% set attr = attr|merge({'data-prototype-name': prototype.vars.name}) %}
    {% endif %}
    {% set attr = attr|merge({'data-allow-add': allow_add ? 1 : 0}) %}
    {% set attr = attr|merge({'data-allow-remove': allow_delete ? 1 : 0 }) %}
    {% set attr = attr|merge({'data-name-prefix': full_name}) %}

    <fieldset class="well">
        <label>{{ form_label(form) }}</label>

        {{ form_errors(form) }}

        {# Don't forget to add the collection attributes in your markup #}
        <table {{ block('widget_attributes') }}>
            <thead>
                <th>Mininal bought quantity</th>
                <th>Maximal bought quantity</th>
                <th>Discount (in percent)</th>
                <th>&nbsp;</th>
            </thead>
            <tbody>

                {#
                 # we can't form_widget(form) as it would render parent markup for a collection, so
                 # we iterate manually on children
                 #}
                {% for price in form %}
                    {{ form_widget(price) }}
                {% endfor %}

            </tbody>
        </table>
    </fieldset>

    {% endspaceless %}
{% endblock %}

{% block PriceType_widget %}

    {# widget_attributes will generate class="item" from the DiscountCollectionType.entry_options configuration #}
    <tr {{ block('widget_attributes') }}>
        <td>
            {{ form_widget(form.quantityFrom, {'attr': {'class': 'form-control'}}) }}
        </td>
        <td>
            {{ form_widget(form.quantityTo, {'attr': {'class': 'form-control'}}) }}
        </td>
        <td>
            {{ form_widget(form.bestUnitPrice, {'attr': {'class': 'form-control'}}) }}
        </td>
        <td>
            {{ form_widget(form.getPaid, {'attr': {'class': 'form-control'}}) }}
        </td>
        <td>
            {{ form_widget(form.vatRate, {'attr': {'class': 'form-control'}}) }}
        </td>
        <td class="text-center">
            <a href="#" class="collection-remove btn btn-default" title="Delete element"><span class="glyphicon glyphicon-trash"></span></a>
            <a href="#" class="collection-add btn btn-default" title="Add element"><span class="glyphicon glyphicon-plus-sign"></span></a>
            <a href="#" class="collection-duplicate btn btn-default" title="Duplicate element"><span class="glyphicon glyphicon-th-large"></span></a>
        </td>
    </tr>

{% endblock %}

{% block PriceType_label %}{% endblock %}
```

```
->add('prices', CollectionType::class, array(
                'entry_type'    => ProductPriceType::class,
                'entry_options' => [
                    'attr' => [
                        'class' => 'price',
                    ],
                ],
                'allow_add'     => true,
                'allow_delete'  => true,
                'attr' => array(
                  'class' => 'table collection-prices'
                ),
                'prototype'    => true,
                'by_reference' => false,
                'delete_empty' => true,
            ))
```

```
<table class="table table-stripped table-bordered">
                                            <thead>
                                                <tr>
                                                    <th>
                                                        Quantity from
                                                    </th>
                                                    <th>
                                                        Quantity to
                                                    </th>
                                                    <th>
                                                        Best unit price
                                                    </th>
                                                    <th>
                                                        You get paid
                                                    </th>
                                                    <th>
                                                        VAT rate
                                                    </th>
                                                    <th>
                                                        Actions
                                                    </th>
                                                </tr>
                                            </thead>
                                            <tbody id="prices_table">
                                                {{ form_row(form.prices) }}
                                            </tbody>
                                        </table>
```

Which selector can I use ?

Thanks for you help

**Comment by ninsuo** [comment]
`$('.table collection-prices').collection({...});`

 I think you meant:

`$('.table.collection-prices').collection({...});`

**Comment by razbounak** [comment]
No, It does not work

**Comment by razbounak** [comment]
Am I obliged to add the .table ?

**Comment by ninsuo** [comment]
You should debug a bit, this error is explicit and tells you that `$('.table collection-prices').length` returns 0. 

I'm working on another project today, I can't help further.

Good luck!

**Comment by ninsuo** [comment]
Yes you should add `table` on the form type, but not necessarily on on the jquery part (the demo doesn't do it)

**Comment by razbounak** [comment]
It's works but my collection is again above my table like before :(

**Comment by ninsuo** [comment]
Yes, this is because you should overwrite all form themes that would create extra markup.

Look at the table using a debugger, you'll see that your `<tr>s` are wrapped inside `<div>` and your browser can't interpret them. You should make sure no extra markup is generated.

**Comment by razbounak** [comment]
I have add 
```
allow_up: false,
allow_down: false,
```

But the <div class="collection-actions"></div> stay here and empty
It's maybe the problem ?

**Comment by ninsuo** [comment]
Hi, 

As soon as the demo works, there should be something wrong in your code.

I'm sorry I am very busy for the whole week, I won't be able to help.

You'd better use regular div grids for your form, it will for sure be easier.

Good luck

**Comment by razbounak** [comment]
I have finally find the solution !
I had left the "jquery.collection.html.twig form theme"

**Comment by razbounak** [comment]
And now, everything works in JS but my collection is not in my entity :(
There is nothing in my collection when I dump $product, $request or $_POST...

**Comment by ninsuo** [comment]
The issue I mentioned regarding the parent of the collection elements is implemented in #77 
Demo at: https://symfony-collection.fuz.org/symfony3/basic/inATable

Cheers
